url-updater/src/main/java/com/devonfw/tools/ide/url/folderhandling/UrlChecksum.java

```diff
@@ -0,0 +1,129 @@
+package com.devonfw.tools.ide.url.folderhandling;
+
+import com.devonfw.tools.ide.url.folderhandling.abstractUrlClasses.AbstractUrlFile;
+
+import java.io.*;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class UrlChecksum extends AbstractUrlFile {
+
+    final HttpClient client = HttpClient.newBuilder().build();
+    final String hashAlgorithm = "SHA-256";
+    private String checksum;
+
+    /**
+     * The constructor.
+     *
+     * @param parent the {@link #getParent() parent folder}.
+     * @param name   the {@link #getName() filename}.
+     */
+    public UrlChecksum(UrlVersion parent, String name) {
+        super(parent, name + ".sha256");
+    }
+
+    public String getChecksum() {
+        return checksum;
+    }
+
+    public void setChecksum(String checksum) {
+        this.checksum = checksum;
+    }
+
+    /**
+     * @param url the url of the download file
+     * @return the input stream of requested url
+     */
+    private InputStream doGetResponseInputStream(String url) {
+        try {
+            HttpRequest request1 = HttpRequest.newBuilder()
+                    .uri(URI.create(url))
+                    .GET()
+                    .build();
+            return client.send(request1, HttpResponse.BodyHandlers.ofInputStream()).body();
+        } catch (IOException | InterruptedException exception) {
+            throw new IllegalStateException("Failed to retrieve response body from url: " + url, exception);
+        } catch (IllegalArgumentException e) {
+            System.out.println("Error while getting response body from url {}\", url, e");
+            return null;
+        }
+    }
+
+    /**
+     * @param inputStream the input stream of requested url
+     * @return checksum of input stream as string
+     */
+    private String doGenerateChecksumFromInputStream(InputStream inputStream)  {
+
+        try {
+            MessageDigest md = MessageDigest.getInstance(hashAlgorithm);
+
+            byte[] buffer = new byte[8192];
+            int bytesRead;
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                md.update(buffer, 0, bytesRead);
+            }
+            inputStream.close();
+
+            byte[] digestBytes = md.digest();
+            String checksum = toHexString(digestBytes);
+            System.out.println(checksum);
+            return checksum;
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to read input stream " + e);
+        } catch (NoSuchAlgorithmException e) {
+            System.out.println("No such hash algorithm " + hashAlgorithm);
```

Comment:
Also throw an `IllegalStateException` instead of returning `null`. In this case the JDK is broken, and we should react and fix the problem rather than trying to continue...

Reply:
done